Guides/Error-Handling.md

```diff
@@ -62,6 +62,16 @@ The `RuntimeError` cases are as follows:
     - See [the official MongoDB documentation](https://docs.mongodb.com/manual/core/read-preference-mechanics/) for more information.
 
 
+### Error Labels
+Some types of errors may contain more specific information describing the context in which they occured. Such errors conform to the `LabeledError` protocol, and the extra information is conveyed through the `errorLabels` property. Specifically, any server error or connection related error may contain labels.
+
+The following error labels are currently defined. Future versions of MongoDB may introduce new labels:
+- `TransientTransactionError`:
```

Comment:
the convenient API handles the retry logic described for these two labels right? might be worth mentioning that users should take a look at `withTransaction`

Reply:
Should I add references to `withTransaction` when it hasn't yet been pushed to master?

Reply:
given SWIFT-776 about documenting why that is preferable, I think we can just leave it out here actually